zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -238,8 +238,13 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
         cnxn.close(ServerCnxn.DisconnectReason.SSL_AUTH_FAILURE);
       }
     } else {
-      // Assign Auth Id according to domains
-      domains.stream().forEach(d -> newAuthIds.add(new Id(getScheme(), d)));
+      domains.forEach(d -> {
```

Comment:
It would be great if you could add some descriptive comments here for this case.